### PR TITLE
feat: support small buttons in chat bubble

### LINKED ISSUE #684: Support small buttons in chat bubble
Casibase could add post-answer functionality support, similar to ChatGPT, including features like answer copying, liking, and disliking.

image

### PR DESCRIPTION
**Pull request by zjuldt**
fix: https://github.com/casibase/casibase/issues/684
copy-button: copy pure-content into clipboard；
like-button & dislike-button: just alert a message now

### CODE REVIEW

File: web/src/ChatBox.js [hunk 1]
```diff
@@ -177,6 +194,27 @@ class ChatBox extends React.Component {
                   <Message.CustomContent>
                     {this.renderMessageContent(message)}
                   </Message.CustomContent>
+                  {
+                    (message.author === "AI" && (this.props.disableInput === false || index !== messages.length - 1)) ? (
+                      <Message.Footer>
+                        {<Button icon={
+                          <>
+                            <svg viewBox="0 0 1024 1024" version="1.1" xmlns="http://www.w3.org/2000/svg" width="22" height="22"><path d="M829.568 53.12H960V1024H194.432v-121.344H64V284.48L361.92 0h467.648v53.12z m0 80.896v768.64H279.488v40.448h595.456V134.016h-45.44zM149.056 317.952v503.808h595.456V80.896H397.248L149.12 317.952z" fill="#cdcdcd"></path></svg>
```
Comment (hsluoyz) [inline web/src/ChatBox.js]: Use image files

File: web/src/ChatBox.js [hunk 2]
```diff
@@ -145,6 +145,23 @@ class ChatBox extends React.Component {
     return message.html;
   };
 
+  copyMessage(message) {
+    const tempElement = document.createElement("div");
+    tempElement.innerHTML = message;
```
Comment (hsluoyz) [inline web/src/ChatBox.js:153]: Use util function for copy, either in this repo or Casdoor

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: web/src/ChatBox.js (hunk 2 above)
Comment (zjuldt) [reply web/src/ChatBox.js:153]: I found the util function is Casdoor(Setting.js), and the code do not work here:
the code is like this :
```
export function getClickable(text) {
  return (
    <a onClick={() => {
      copy(text);
      showMessage("success", "Copied to clipboard");
    }}>
      {text}
    </a>
  );
}
```
and I think write a new function to Parse HTML into clipboard is better;
new function is like this : 
```
  copyMessageFromHTML(message) {
    const tempElement = document.createElement("div");
    tempElement.innerHTML = message;
    const text = tempElement.innerText;
    copy(text);
    Setting.showMessage("success", "Message copied successfully");
  }
```

File: web/src/ChatBox.js (hunk 1 above)
Comment (zjuldt) [reply web/src/ChatBox.js]: changed to use **ant-design icon**;
```
import {CopyOutlined, DislikeOutlined, LikeOutlined} from "@ant-design/icons";
```
```
{<Button icon={<CopyOutlined />} style={{border: "none"}} onClick={() => this.copyMessageFromHTML(message.html.props.dangerouslySetInnerHTML.__html)}></Button>}
{<Button icon={<LikeOutlined />} style={{border: "none"}} onClick={() => this.likeMessage()}></Button>}
{<Button icon={<DislikeOutlined />} style={{border: "none"}} onClick={() => this.dislikeMessage()}></Button>}
```

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**